app/serializers.py

```diff
@@ -0,0 +1,64 @@
+from rest_framework import serializers
+
+from app.models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for ingredients."""
+
+    class Meta:
+        model = Ingredient
+        fields = '__all__'
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for recipes."""
+
+    ingredients = IngredientSerializer(many=True)
+
+    class Meta:
+        model = Recipe
+        fields = '__all__'
+
+    def _get_or_create_ingredients(self, ingredients, recipe):
+        """Handle getting or creating ingredients as needed."""
+        for ingredient in ingredients:
+            ingredient_obj, created = Ingredient.objects.get_or_create(
+                **ingredient
+            )
+            recipe.ingredients.add(ingredient_obj)
+
+    def create(self, validated_data):
```

Comment:
Just as information, we don't use in tk apps the serializer for creating, updating or deleting data. For this project it's totally ok because that's what the course suggests

Reply:
Okay thanks for the heads up.